test/spec_server.rb

```diff
@@ -141,7 +141,8 @@ def with_stderr
   it "check pid file presence and running process" do
     pidfile = Tempfile.open('pidfile') { |f| f.write($$); break f }.path
     server = Rack::Server.new(:pid => pidfile)
-    server.send(:pidfile_process_status).must_equal :running
+    server.send(:pidfile_process_status).must_equal :this_process
+    `ruby -rrack/server -e 'print Rack::Server.new(:pid => "#{pidfile}").send(:pidfile_process_status).inspect'`.must_equal ":running"
```

Comment:
I dug around stdlib and special variables for an actual C argv[0]. Do you have any idea where it's at?